crates/holochain_state/src/source_chain/source_chain_buffer.rs

```diff
@@ -176,46 +192,46 @@ impl SourceChainBuf {
     }
 
     /// dump the entire source chain as a pretty-printed json string
-    pub async fn dump_as_json(&self) -> Result<String, SourceChainError> {
-        #[derive(Serialize, Deserialize)]
-        struct JsonElement {
-            pub signature: Signature,
-            pub header_address: HeaderHash,
-            pub header: Header,
-            pub entry: Option<Entry>,
-        }
-
-        // TODO fix this.  We shouldn't really have nil values but this would
-        // show if the database is corrupted and doesn't have an element
-        #[derive(Serialize, Deserialize)]
-        struct JsonChainDump {
-            element: Option<JsonElement>,
-        }
-
+    pub async fn dump_as_json(&self) -> Result<SourceChainJsonDump, SourceChainError> {
```

Comment:
The name seems misleading now that this returns a serialized struct instead of already serialized JSON